core/src/main/java/org/infinispan/configuration/cache/BackupConfigurationBuilder.java

```diff
@@ -148,6 +148,9 @@ public void validate() {
          throw new CacheConfigurationException("It is required to specify a 'failurePolicyClass' when using a " +
                                                 "custom backup failure policy!");
       }
+      if (useTwoPhaseCommit && strategy == BackupConfiguration.BackupStrategy.ASYNC) {
+         throw new CacheConfigurationException("'two-phase-commit' can be used only with SYNC backup strategy!");
```

Comment:
I'm still not sure what would be the best option... it would probably be clearer to throw the exception for the ASYNC/2PC combination, but I we did support it in 7.0.x. @tristantarrant WDYT?

WRT the exception, we did start i18n-ing some of the configuration builder exceptions (e.g. InvocationBatchingConfigurationBuilder), so I would vote for changing all the exceptions in the same file.